tests/lib/sample_rate_converter/src/main.c

```diff
@@ -0,0 +1,1065 @@
+		      "Input sample rate not as expected %d", conv_ctx.input_sample_rate);
+	zassert_equal(conv_ctx.output_sample_rate, output_sample_rate,
+		      "Output sample rate not as expected");
+	zassert_equal(conv_ctx.conversion_ratio, conversion_ratio,
+		      "Conversion ratio not as expected");
+	zassert_equal(conv_ctx.conversion_direction, CONVERSION_DIR_UP,
+		      "Conversion direction is not as expected");
+	zassert_equal(conv_ctx.filter_type, filter, "Filter set incorrectly");
+	zassert_equal(conv_ctx.input_buf.bytes_in_buf, 0,
+		      "Bytes in input buffer not as expected (%d)",
+		      conv_ctx.input_buf.bytes_in_buf);
+	zassert_equal(ring_buf_size_get(&conv_ctx.output_ringbuf), 24,
+		      "Number of bytes in output ringbuffer not as expected %d",
+		      ring_buf_size_get(&conv_ctx.output_ringbuf));
+
+	zassert_equal(output_written, expected_output_samples * sizeof(uint32_t),
+		      "Output size was not as expected (%d)", output_written);
+
+	/* Verify output bytes, expected three samples in output per input */
+	for (int i = 0; i < expected_output_samples; i++) {
+		zassert(output_samples[i] < input_one[(i / 3)],
+			"Output samples is not smaller than corresponding input");
+	}
+
+	/* Second run */
+	ret = sample_rate_converter_process(
+		&conv_ctx, filter, input_two, num_samples * sizeof(uint32_t), input_sample_rate,
+		output_samples, expected_output_samples * sizeof(uint32_t), &output_written,
+		output_sample_rate);
+
+	zassert_equal(ret, 0, "Sample rate conversion process failed");
+	zassert_equal(conv_ctx.input_sample_rate, input_sample_rate,
+		      "Input sample rate not as expected %d", conv_ctx.input_sample_rate);
+	zassert_equal(conv_ctx.output_sample_rate, output_sample_rate,
+		      "Output sample rate not as expected");
+	zassert_equal(conv_ctx.conversion_ratio, conversion_ratio,
+		      "Conversion ratio not as expected");
+	zassert_equal(conv_ctx.conversion_direction, CONVERSION_DIR_UP,
+		      "Conversion direction is not as expected");
+	zassert_equal(conv_ctx.filter_type, filter, "Filter set incorrectly");
+	zassert_equal(conv_ctx.input_buf.bytes_in_buf, 4,
+		      "Bytes in input buffer not as expected %d", conv_ctx.input_buf.bytes_in_buf);
+	zassert_equal(ring_buf_size_get(&conv_ctx.output_ringbuf), 12,
+		      "Number of bytes in output ringbuffer not as expected %d",
+		      ring_buf_size_get(&conv_ctx.output_ringbuf));
+
+	uint32_t expected_input_buf_two[] = {8000};
+	zassert_mem_equal(conv_ctx.input_buf.buf, expected_input_buf_two, 1 * sizeof(uint32_t));
+
+	zassert_equal(output_written, expected_output_samples * sizeof(uint32_t),
+		      "Output size was not as expected (%d)", output_written);
+
+	/* Verify output bytes, expected three samples in output per input */
+	for (int i = 0; i < expected_output_samples; i++) {
+		zassert(output_samples[i] < input_two[(i / 3)],
+			"Output samples is not smaller than corresponding input");
+	}
+
+	/* Third run */
```

Comment:
Several tests have multiple runs which is good, but can we execute in a loop?

Reply:
We probably could, but we would have to keep track of at least three values that change between each iteration, so I think the test case is clearer without additional loop-logic